zeyple/zeyple.py

```diff
@@ -83,6 +89,78 @@ def gpg(self):
 
+    def _validate_dnssec(self, domain):
+        # if force_dnssec is enabled go ahead
+        if self.config.has_option('zeyple', 'force_dnssec') and \
+            self.config.getboolean('zeyple', 'force_dnssec'):
+
+            dnssec_check = urllib.urlopen("http://portfolio.sidnlabs.nl:8080/check/" + domain).read().split(',')
+            if dnssec_check[2] == "secure":
+                logging.info("DNSSec: " + domain + " is secure")
+                return True
+            else:
+                logging.warn("DNSSec: " + domain + " is insecure")
+                return False
+        else:
+            return True
+
+    def _retrieve_pka_key(self, recipient):
+        username,domainname = recipient.split("@")
+
+        # if use_pka is is enabled go ahead
+        if self.config.has_option('zeyple', 'use_pka') and \
+            self.config.getboolean('zeyple', 'use_pka'):
+
+            username,domainname = recipient.split("@")
+            qname = "%s._pka.%s"%(username,domainname)
+
+            # if _validate_dnssec returns false (dnssec invalid) return false
+            if not (self._validate_dnssec(domainname)):
+                return False
+            else:
+                try:
+                    answer=myResolver.query(qname, "TXT")
+                    for rdata in answer:
+                        for txt_string in rdata.strings:
+                            temp_pka=(txt_string.replace(';','=')).lower()
+                            pka=temp_pka.split("=")
+                            for n,i in enumerate(pka):
+                                # PKA fingerprint
+                                   if pka[n] == "fpr":
+                                       pka_fpr = (pka[n+1]).lower()
+                                   if pka[n] == "uri":
+                                       logging.info("PKA: " + "['" + qname + "'] returned ['" + pka[n+1] + "']")
+                                       sock = urllib.urlopen(pka[n+1])
+                                       key_data=sock.read()
+                                       sock.close()
+                                       # gnupg library
+                                       gpg = gnupg.GPG(gnupghome=self.config.get('gpg', 'home'))
+                                       import_result = gpg.import_keys(key_data)
+
+                                       # Double check PKA and key fingerprint. python-gnupg > 0.3.7 support gpg.scan_keys method to retrieve fingerprint before import
+                                       fingerprint = []
+                                       key_fpr = (import_result.fingerprints[0]).lower()
+
+                                       if pka_fpr == key_fpr:
+                                           # logging.info("PKA: fingerprint ['" + pka_fpr + "'] = Key fingerprint ['" + key_fpr + "']")
+                                           logging.info("PKA: PKA and Key fingerprint match. Key for ['" + recipient + "'] imported")
```

Comment:
This lines are getting very very long, and very very nested. The [zen of python](https://www.python.org/dev/peps/pep-0020/) says "Flat is better than nested."